pages/common/xidel.md

```diff
@@ -0,0 +1,36 @@
+# xidel
+
+> A command line tool to download and extract data from HTML/XML pages as well as JSON APIs.
+> More information: <https://www.videlibri.de/xidel.html>.
+
+- Print all URLs found by a google search:
+
+`xidel https://www.google.com/search?q=test --extract "//a/extract(@href, 'url[?]q=([^&]+)&', 1)[. != '']"`
+
+- Print the title of all pages found by a google search and download them:
+
+`xidel {{https://www.google.com/search?q=test}} --follow "{{//a/extract(@href, 'url[?]q=([^&]+)&', 1)[. != '']}}" --extract {{//title}} --download {{'{$host}/'}}`
+
+- Follow all links on a page and print the titles, With XPath:
+
+`xidel {{https://example.org}} -f {{//a}} -e {{//title}}`
+
+- Follow all links on a page and print the titles, With CSS selectors:
+
+`xidel {{https://example.org}} -f "{{css('a')}}" --css {{title}}`
+
+- Follow all links on a page and print the titles, With pattern matching:
+
+`xidel {{https://example.org}} -f "{{<a>{.}</a>*}}" -e "{{<title>{.}</title>}}"`
+
+- If you have an example.xml file, you can read the important part like (which will also check, if the element containing "ood" is there, and fail otherwise):
+
+`xidel {{example.xml}} -e "{{<x><foo>ood</foo><bar>{.}</bar></x>}}"`
```

Comment:
```suggestion
- Read the pattern from example.xml (which will also check if the element containing "ood" is there, and fail otherwise):

`xidel {{path/to/example.xml}} -e "{{<x><foo>ood</foo><bar>{.}</bar></x>}}"`
```